homework9.py

```diff
@@ -0,0 +1,102 @@
+""" Homework9: Functions writing and implementing programs
+in Python """
+
+# Validate
+# Ваша задача написать программу, принимающее число - номер кредитной карты
+# (число может быть четным или не четным).
+# И проверяющей может ли такая карта существовать. Предусмотреть защиту от
+# ввода букв, пустой строки и т.д. Примечания Алгоритм Луна.
+
+from functools import reduce
+
+
+def validate_card(card_number):
+    """
+    Validate a credit card number using the Luhn algorithm.
+    """
+    even = []
+    odd = []
+    card_number_reverted = card_number[::-1]
+    for elem in range(len(card_number)):
+        if (elem + 1) % 2 == 0:
+            odd_number = int(card_number_reverted[elem]) * 2
+            if odd_number > 9:
+                odd_number -= 9
+            odd.append(odd_number)
+
+        else:
+            even.append(card_number_reverted[elem])
+
+    result = reduce(lambda x, y: x + y, [*list(map(int, even)),
+                                         *odd])
+    if result % 10 == 0:
+        return True
+    return False
+
+
+print(validate_card('4561261212345464'.replace(' ', '')))
+print(validate_card('4561261212345467'.replace(' ', '')))
+
+
+# Последовательность
+# Дана последовательность целых чисел в виде массива. Определите, можно ли
+# получить строго возрастающую последовательность, удалив из массива не более
+# одного элемента.
+# Примечание: последовательность a0, a1, ..., an считается строго возрастающей,
+# если a0 < a1 < ... < an. Последовательность, содержащая только один элемент,
+# также считается строго возрастающей.
+# Примеры
+# Для последовательности = [1, 3, 2, 1], вывод должен быть решение = False.
+# В этом массиве нет ни одного элемента, который можно было бы удалить, чтобы
+# получить строго возрастающую последовательность.
+# Для последовательности = [1, 3, 2] вывод должен быть = True.
+# Вы можете удалить 3 из массива, чтобы получить строго возрастающую
+# последовательность [1, 2]. Альтернативно можно убрать 2, чтобы получить
+# строго возрастающую последовательность [1, 3].
+
+
+def can_be_ascending_by_removing_one(seq_int):
+    """
+    Get a strictly ascending sequence by removing only one element.
+    """
+    count = 0
+
```

Comment:
before 'for' loop we might use check with set, something like     
if (len(seq_int) - len(set(seq_int))) >1:
        print('duplicate')
it provides us opportunity to skip for loop if our array contains more than two duplicated numbers, if we have two it means that we are not able to remove one and have ascending array
for instance: [1,2,1,2] even if remove one any element, we still have duplicate.
So, my idea is to add that verification as well as write explanation why it here (before returning False)

Reply:
please check my corrections